arbnode/transaction_streamer.go

```diff
@@ -1202,11 +1242,13 @@ func (s *TransactionStreamer) ExecuteNextMsg(ctx context.Context) bool {
 		return false
 	}
 
-	msgWithBlockHash := arbostypes.MessageWithMetadataAndBlockHash{
-		MessageWithMeta: msgAndBlockHash.MessageWithMeta,
+	msgWithBlockInfo := arbostypes.MessageWithMetadataAndBlockInfo{
+		MessageWithMeta: msgAndBlockInfo.MessageWithMeta,
 		BlockHash:       &msgResult.BlockHash,
+		// maybe if blockhash is differing we clear out previous timeboosted and not send timeboosted info to broadcasting?
```

Comment:
Cool. I think the best place to implement this is in [Bulk syncing of missing BlockMetadata](https://github.com/OffchainLabs/nitro/pull/2765) PR, since it tracks missing blockMetadata and in there we can clear out blockMetadata and mark it as missing in DB so that it will be refetched later on